Matrix trimming can cause failure in "Download Build Artifact(s)" step

It is possible for matrix trimming to trim out all Dockerfiles from a build. In that case, the remaining stages of the build should essentially no-op. However, in this scenario, it is possible for the ["Download Build Artifact(s)" step](https://github.com/dotnet/docker-tools/blob/5c0037021e0d557350faeaef7808e4b97cddcca8/eng/common/templates/jobs/post-build.yml#L17-L19) of the Post Build stage to fail with the following error:

```
##[error]Could not find any pipeline artifacts in the build.
```

There is actually a race condition going on here. This step can succeed as long as the `SDLSources` stage has published its pipeline artifacts. But if that has not happened by the time the download step executes, then it will fail. Normally, build jobs would be producing their own artifacts but that doesn't happen in this scenario because of the trimming.

[Example build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2564022&view=results) (internal link)

--- a/eng/common/templates/jobs/post-build.yml
+++ b/eng/common/templates/jobs/post-build.yml
@@ -17,6 +17,8 @@ jobs:
   - template: /eng/common/templates/steps/download-build-artifact.yml@self
     parameters:
       targetPath: $(Build.ArtifactStagingDirectory)
+      # This can fail if no build jobs ran to produce any artifacts
+      continueOnError: true
   - powershell: |
       # Move all image-info artifacts to their own directory
       New-Item -ItemType Directory -Path $(imageInfosHostDir)

--- a/eng/common/templates/steps/download-build-artifact.yml
+++ b/eng/common/templates/steps/download-build-artifact.yml
@@ -2,6 +2,7 @@ parameters:
   targetPath: ""
   artifactName: ""
   condition: true
+  continueOnError: false
 
 steps:
 - task: DownloadPipelineArtifact@1
@@ -15,3 +16,4 @@ steps:
     artifactName: ${{ parameters.artifactName }}
   displayName: Download Build Artifact(s)
   condition: and(succeeded(), ${{ parameters.condition }})
+  continueOnError: ${{ parameters.continueOnError }}

